libc/src/__support/FPUtil/dyadic_float.h

```diff
@@ -245,22 +246,23 @@ template <size_t Bits> struct DyadicFloat {
     StorageType result =
         FPBits::create_value(sign, out_biased_exp, out_mantissa).uintval();
 
-    switch (quick_get_round()) {
-    case FE_TONEAREST:
-      if (round && (lsb || sticky))
-        ++result;
-      break;
-    case FE_DOWNWARD:
-      if (sign.is_neg() && (round || sticky))
-        ++result;
-      break;
-    case FE_UPWARD:
-      if (sign.is_pos() && (round || sticky))
-        ++result;
-      break;
-    default:
-      break;
-    }
+    if constexpr (!cpp::is_constant_evaluated())
+      switch (quick_get_round()) {
```

Comment:
We always need to round, it's just that in `constexpr` contexts we use `FE_TONEAREST` as rounding mode.

```suggestion
    int rounding_mode = FE_TONEAREST;
    if constexpr (!cpp::is_constant_evaluated())
      rounding_mode = quick_get_round();

    switch (rounding_mode) {
```